### PR TITLE
fix(order): skip non-positive match candidates

### PR DESCRIPTION
**Pull request by phroi**
## Why

`OrderManager.bestMatch` counted non-positive-gain partial candidates as rejected, but the same candidate could still become the returned best match when the empty frontier was allowance-rejected. That allowed a match with zero or negative economic gain to be selected.

## Changes

- Keep match-search frontier advancement separate from the returned best profitable match.
- Leave no-match diagnostics at `bestGain: 0n` instead of the sentinel value.
- Add a regression test covering a non-positive candidate after an allowance-rejected empty frontier.
- Align the exhaustive test oracle with the production invariant that non-empty selected matches must have positive gain.

### CODE REVIEW

File: packages/order/src/order.ts [hunk 1]
```diff
@@ -387,22 +387,24 @@ export class OrderManager implements ScriptDeps {
...
     let best = {
-      i: -1,
-      j: -1,
       ckbDelta: 0n,
       udtDelta: 0n,
       partials: [] as Match["partials"],
       ckbAllowance: allowance.ckbValue,
       udtAllowance: allowance.udtValue,
+      gain: 0n,
+    };
```
[the remark this answers is not in the document]
Comment (phroi) [reply packages/order/src/order.ts:394]: Removed the stale `ckbAllowance` and `udtAllowance` fields from the tracked best match. The local loop variables still own allowance checks; `best` now stores only returned match data plus gain.

File: packages/order/src/order.test.ts [hunk 2]
```diff
@@ -1763,13 +1813,18 @@ function exhaustiveSequentialBestMatch(
       const udtAllowance = allowance.udtValue + udtDelta;
       const gain = (ckbDelta - ckbFee) * exchangeRate.ckbScale + udtDelta * exchangeRate.udtScale;
 
-      if (ckbAllowance >= 0n && udtAllowance >= 0n && gain > bestGain) {
+      if (
+        ckbAllowance >= 0n &&
+        udtAllowance >= 0n &&
+        (partials.length === 0 || gain > 0n) &&
+        gain > bestGain
+      ) {
```
[the remark this answers is not in the document]
Comment (phroi) [reply packages/order/src/order.test.ts:1820]: Simplified the oracle condition. With `bestGain` initialized to `0n`, `gain > bestGain` already excludes zero and negative candidates.

**Comment by phroi** [comment]
LGTM

Phroi %37
